src/styles/index.scss

```diff
@@ -1,3 +1,56 @@
+@import '/src/styles/variables';
+
 body {
+  font-family: Arial, Helvetica, sans-serif;
+  font-size: 30px;
   margin: 0;
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  height: 100vh;
+}
+
+.calendar {
+  display: flex;
+  flex-wrap: wrap;
+  width: ($size-day * 7) + ($size-gap * 6);
+  gap: $size-gap;
+
+  &__day {
+    display: flex;
+    height: $size-day;
+    width: $size-day;
+    box-sizing: border-box;
+    align-items: center;
+    justify-content: center;
+    background-color: #eee;
+    border: 1px solid black;
+    transition: all 500ms;
+
+    &:hover {
+      cursor: pointer;
+      background-color: #ffbfcb;
+      transform: translateY(-20px);
+    }
+
+    @for $i from 1 through 31 {
+      &:nth-child(#{$i})::before {
+        content: '#{$i}';
+      }
+    }
+  }
+}
+
+@for $i from 28 through 31 {
+  .calendar--month-length-#{$i} .calendar__day:nth-child(n + #{$i + 1}) {
+    display: none;
+  }
+}
+
+@each $name-day, $nomber-dey in $day {
+  .calendar--start-day-#{$name-day} {
+    .calendar__day:first-child {
+      margin-left: ($nomber-dey - 1) * ($size-day + $size-gap);
+    }
+  }
 }
```

Comment:
This loop is used to set the start day of the month. However, it's better to control this in your JavaScript code. This way, you can avoid unnecessary DOM manipulations and improve the performance of your application.